### DISCUSSION TITLE
linear exposure

### DISCUSSION THREAD

**Discussion by joelRVC** [ggarra13/mrv2#311] [Ideas]
As far as I have seen from the program, there is no way to change the exposure linearly. I usually view EXR and use the OCIO views to apply DRT's to the sequences. On certain occasions it is necessary to increase/lower the exposure for testing, but the exposure and many others Color control options only seem to be applied after OCIO, applied in this context it already loses the purpose for which OCIO is used in the first place.
My suggestion in that case is if you can add an option in preferences to define whether the exposure is applied linearly (before the view), or you can add it directly in the F-stops UI (with better planning so that it does not disturb the current workflow of course).

**Comment by ggarra13** [comment]
> As far as I have seen from the program, there is no way to change the exposure linearly. 

The UI control for the gain in the main topbar is indeed in log and it indeed works like in Nuke, or at least how it used to work in Nuke. 

>I usually view EXR and use the OCIO views to apply DRT's to the sequences. 

Not sure what the DRT acronym stands for.  Can you clarify?

> On certain occasions it is necessary to increase/lower the exposure for testing, but the exposure and many others Color control options only seem to be applied after OCIO, applied in this context it already loses the purpose for which OCIO is used in the first place.

This used to be the case on older versions of mrv2 (a bug in tlRender's code) previous to v1.2.4, but it is no longer the case on recent versions.  Exposure and all other color controls are applied to the image in linear space *BEFORE* OCIO.

> My suggestion in that case is if you can add an option in preferences to define whether the exposure is applied linearly (before the view), or you can add it directly in the F-stops UI (with better planning so that it does not disturb the current workflow of course).

As mentioned, exposure/gain is applied in linear space.  

It is only the UI gain/exposure control that is in log, as that matches human perception more closely.

**Comment by joelRVC** [comment]
> This used to be the case on older versions of mrv2 (a bug in tlRender's code) previous to v1.2.4, but it is no longer the case on recent versions

Oh I see, my bad, I was testing it with the test version that you sent me before and it did behave as before, I am testing the official release version and it does work correctly as you mentioned, thank you.

> The UI control for the gain in the main topbar is indeed in log and it indeed works like in Nuke, or at least how it used to work in Nuke.

Oh that makes more sense. I come from a background of using more 3D without much real footage and there we usually use linear controls, that's why it wasn't clear to me the first time.

> Not sure what the DRT acronym stands for. Can you clarify?

Stands for Display Rendering Transform, is a more current term that more objectively describes the final part of the image formation chain: visualization. It is another way of referring to the views of the color config. A piece from Chris Brejon's site describes this better:

![{BFC06B76-133F-47BD-B925-FC2B0B48DB96}](https://github.com/user-attachments/assets/4ba89c3b-19cf-4b59-a40e-4d664c497a97)
